jbpm/jbpm-flow/src/main/java/org/jbpm/process/core/context/exception/ExceptionScope.java

```
@@ -41,35 +49,51 @@ public void setExceptionHandler(String exception, ExceptionHandler exceptionHand
         this.exceptionHandlers.put(exception, exceptionHandler);
     }
 
-    public ExceptionHandler getExceptionHandler(String exception) {
-        ExceptionHandler result = exceptionHandlers.get(exception);
-        if (result == null) {
-            result = exceptionHandlers.get(null);
-        }
-        return result;
+    private Throwable getException(KogitoProcessContext context) {
+        return (Throwable) context.getContextData().get("Exception");
     }
 
-    public ExceptionHandler getExceptionHandler(Throwable exception) {
-        Class<?> exceptionClass = exception.getClass();
-        ExceptionHandler handler = exceptionHandlers.get(exceptionClass.getName());
+    private boolean isHandler(ExceptionHandler handler, Throwable exception) {
+        return handler.getExceptionCode().map(code -> test(code, exception)).orElse(false);
+    }
 
-        if (handler == null) {
-            Collection<ExceptionHandlerPolicy> policies = ExceptionHandlerPolicyFactory.getHandlerPolicies();
-            handler = exceptionHandlers.entrySet().stream().filter(e -> test(policies, e.getKey(), exception)).findFirst().map(Entry::getValue).orElse(null);
+    public ExceptionHandler getExceptionHandler(String key, Throwable exception, KogitoProcessContext context) {
+        Optional<ExceptionHandler> result = Optional.empty();
+        if (key != null) {
+            result = Optional.ofNullable(exceptionHandlers.get(key));
+        } else if (exception != null) {
+            Optional<String> errorName = getErrorName(context);
+            if (errorName.isPresent()) {
+                return errorName.map(k -> exceptionHandlers.get(k)).filter(h -> isHandler(h, exception)).orElse(null);
```

Why this time we don't handle through `result` variable thus omitting the fallback `return result.orElse(exceptionHandlers.get(null))` ?

---

This is intentional, when errorName (which is a new addition of this PR for SWF) is present, if not specific handler for that name is found (or that handler does not include an errorCode) we basically disable bpmn error handling and let the flow fail